From d0f66bfc8320f9f17f24e8682d34d283b2bc568c Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Fri, 13 Aug 2021 08:13:21 +0300
Subject: [PATCH] [SW-36119] preboot: gaudi2: print strap settings

System team requested to print Gaudi2 boot strap pin values at boot
It is aligned to print them at preboot stage

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I6ec2748977a31361a4d520f01ef02d2128448fd2
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 1800d81..1221a5b 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -372,6 +372,7 @@
 __weak void preboot_device_id_init(void) {}
 __weak u32 preboot_device_id_get(void) { return 0; }
 __weak void preboot_engines_init(void) {}
+__weak void preboot_read_straps(void) {}
 
 void preboot_main(void)
 {
@@ -417,6 +418,7 @@
 	/* Prepare boot FW versions and eFuse data for host */
 	preboot_version();
 	fuse_version();
+	preboot_read_straps();
 	preboot_security_dynamic_vers_init(ver_off, sizeof(preboot_vers));
 
 	preboot_led_on();
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 84a2cfa..44a0e09 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -579,3 +579,16 @@
 	preboot_puts(" reset\n");
 }
 
+void preboot_read_straps(void)
+{
+	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
+
+	preboot_puts_hex_raw("Boot strap pins low ", boot_strap, false);
+	preboot_puts("\n");
+	boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H);
+	preboot_puts_hex_raw("Boot strap pins high ", boot_strap, false);
+	preboot_puts("\n");
+	boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_LEGACY_BOOT_STRAPS);
+	preboot_puts_hex_raw("Boot strap pins legacy ", boot_strap, false);
+	preboot_puts("\n");
+}
